From 24fa0ba264d9e5f263be9082e862dc80305419d4 Mon Sep 17 00:00:00 2001
Message-Id: <24fa0ba264d9e5f263be9082e862dc80305419d4.1430272235.git.chang-joon.lee@intel.com>
In-Reply-To: <0282bce11918ec0dd0c4e7e3bd54e392c2b27a51.1430272235.git.chang-joon.lee@intel.com>
References: <0282bce11918ec0dd0c4e7e3bd54e392c2b27a51.1430272235.git.chang-joon.lee@intel.com>
From: Deepak M <m.deepak@intel.com>
Date: Fri, 17 Apr 2015 18:01:24 +0530
Subject: [PATCH 05/10] FOR_UPSTREAM [VPG]: drm/i915: Programming the lane
 count and tx request

Programming the lane count and tx request in the flis_dsi register.

Issue: GMINL-8141
Change-Id: I870dee49c197497868245ad0152ffd60f6b9544c
Signed-off-by: Yogesh Mohan Marimuthu <yogesh.mohan.marimuthu@intel.com>
---
 drivers/gpu/drm/i915/intel_dsi.c |   15 +++++++++++++++
 drivers/gpu/drm/i915/intel_dsi.h |    1 +
 2 files changed, 16 insertions(+)

diff --git a/drivers/gpu/drm/i915/intel_dsi.c b/drivers/gpu/drm/i915/intel_dsi.c
index dd504d8..5caba02 100644
--- a/drivers/gpu/drm/i915/intel_dsi.c
+++ b/drivers/gpu/drm/i915/intel_dsi.c
@@ -448,6 +448,21 @@ static void intel_dsi_pre_enable(struct intel_encoder *encoder)
 
 	do {
 		I915_WRITE(MIPI_DEVICE_READY(pipe), 0x0);
+		if (intel_dsi->dual_link)
+			pipe = PIPE_B;
+	} while (--count > 0);
+
+	tmp = vlv_flisdsi_read(dev_priv, FLIS_RCOMP_IOSFSB_REG3);
+	tmp = tmp | ((1 << 31) | (4 << 28));
+	vlv_flisdsi_write(dev_priv, FLIS_RCOMP_IOSFSB_REG3, tmp);
+
+	if (intel_dsi->dual_link) {
+		pipe = PIPE_A;
+		count = 2;
+	} else
+		count = 1;
+
+	do {
 		val = intel_dsi->lane_count << DATA_LANES_PRG_REG_SHIFT;
 		val |= intel_dsi->channel << VID_MODE_CHANNEL_NUMBER_SHIFT;
 		val |= intel_dsi->pixel_format;
diff --git a/drivers/gpu/drm/i915/intel_dsi.h b/drivers/gpu/drm/i915/intel_dsi.h
index f3b8c99..cf60073 100644
--- a/drivers/gpu/drm/i915/intel_dsi.h
+++ b/drivers/gpu/drm/i915/intel_dsi.h
@@ -397,6 +397,7 @@
 
 #define LPIO_PWM_ENABLE_MASK		0x80000000
 
+#define FLIS_RCOMP_IOSFSB_REG3		0x03
 #define FLIS_RCOMP_IOSFSB_REG4		0x04
 #define FLIS_DSI_TXCNTRL		0x15
 #define BIT_FILS_DSI_TXCNTRL_HS_IO_SEL	(1 << 0)
-- 
1.7.9.5

